core/trino-main/src/main/java/io/trino/sql/gen/BytecodeUtils.java

```diff
@@ -354,7 +354,7 @@ else if (type == ConnectorSession.class) {
     private static InvocationArgumentConvention getPreferredArgumentConvention(BytecodeNode argument, int argumentCount, boolean nullable)
     {
         // a Java function can only have 255 arguments, so if the count is low use block position or boxed nullable as they are more efficient
-        if (argumentCount <= 100) {
+        if (argumentCount <= 84) {
```

Comment:
I assume this was chosed becuse `256 / 3 = 85.3`.  There are other parameters that can be added, like session.  I would make this 64 instead, which leaves plenty of space for extra stuff.

Reply:
Yes, the number was calculated that way and I tested with the parameters counts around that threshold and all queries compiled correctly